TELEMETRY.md

```diff
@@ -9,83 +9,145 @@
+geo_city
+``` 
+
+### Peculiarities
 
-## Example sequence for a 'voted => not sure' interaction
+1. The places_bookmarks_count_histogram and places_pages_count_histogram attributes are only available occasionally in the telemetry session due to performance reasons. We collect both the histograms as present in the current telemetry session (using the attributes places_bookmarks_count_histogram and places_pages_count_histogram) as well as the actual values that telemetry occasionally would collect. The places_bookmarks_count and places_pages_count attributes will always include the values based on the current database contents.
+1. Contrary to what the [documentation](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/data/main-ping.html) states ("Flag and count histograms are always created and submitted, with their default value being respectively false and 0. Other histogram types (see Choosing a Histogram Type) are not created nor submitted if no data was added to them."), the search_counts histogram is empty until a search has been executed.
+1. Scalars and most histograms are only submitted if data was added to them.
+1. Config/preference-related attributes are generally only defined if the defer from the browser default preferences
+1. Regarding `system_gfx.monitors[1].screen_width`, we want to collect the screen widthof the user's primary monitor. During collection, the primary monitor values are found at `environment.system.gfx.monitors[0].screenWidth` but the assumption is that this value ends up at `system_gfx.monitors[1].screen_width` in Re:dash after data processing. To be able to verify this assumption, the `system_gfx.monitors[1].screen_width_zero_indexed` is also submitted, corresponding to `environment.system.gfx.monitors[1].screenWidth`
+
+## Example sequence
 
 These are the `payload` fields from all pings in the `shield-study` and `shield-study-addon` buckets.
 
 ```
 
 // common fields
 
-branch        esper-pioneer        // should describe Question text
+branch        pioneer
 study_name    esper-pioneer-shield-study
-addon_version 1.0.0
+addon_version 0.0.1
 version       3
 
-2017-10-09T14:16:18.042Z shield-study
+
+2017-10-31T11:35:39.973Z shield-study-addon
 {
-  "study_state": "enter"
+  "attributes": {
+    "app_version": "58.0a1",
+    "default_search_engine": "undefined",
+    "event": "telemetry-payload",
+    "locale": "en-US",
+    "normalized_channel": "nightly",
+    "os": "Darwin",
+    "os_version": "16.7.0",
+    "places_bookmarks_count": "10",
+    "places_bookmarks_count_histogram": "undefined",
+    "places_pages_count": "10",
+    "places_pages_count_histogram": "undefined",
+    "profile_creation_date": "undefined",
+    "profile_subsession_counter": "1",
+    "scalar_parent_browser_engagement_max_concurrent_tab_count": "1",
+    "scalar_parent_browser_engagement_max_concurrent_window_count": "1",
+    "scalar_parent_browser_engagement_navigation_about_newtab": "undefined",
```

Comment:
'undefined' here seems Not Ideal.

Not a ton of options though.  I suspect you want one of these instead:

- `{}`  (pretend it's present but empty)
- `""`  (this will scare `JSON.parse` though)

As you have seen, if a key is undefined, JSON.stringify drops it unless you define a replacer.


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify